Formula/o/opencv.rb

```diff
@@ -76,83 +77,90 @@ def install
-    end
-
-    # Ref: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options
-    ENV.runtime_cpu_detection
-    if Hardware::CPU.intel? && build.bottle?
-      cpu_baseline = MacOS.version.requires_sse42? ? "SSE4_2" : "SSSE3"
-      args += %W[-DCPU_BASELINE=#{cpu_baseline} -DCPU_BASELINE_REQUIRE=#{cpu_baseline}]
-    end
 
-    system "cmake", "-S", ".", "-B", "build_shared", *args, *std_cmake_args
-    inreplace "build_shared/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
-    system "cmake", "--build", "build_shared"
-    system "cmake", "--install", "build_shared"
+      args += [
+        "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON", # https://github.com/protocolbuffers/protobuf/issues/12292
+        "-Dprotobuf_MODULE_COMPATIBLE=ON", # https://github.com/protocolbuffers/protobuf/issues/1931
+      ]
 
-    system "cmake", "-S", ".", "-B", "build_static", *args, *std_cmake_args, "-DBUILD_SHARED_LIBS=OFF"
-    inreplace "build_static/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
-    system "cmake", "--build", "build_static"
-    lib.install buildpath.glob("build_static/{lib,3rdparty/**}/*.a")
+      # Disable precompiled headers and force opencv to use brewed libraries on Linux
+      if OS.linux?
+        args += %W[
+          -DENABLE_PRECOMPILED_HEADERS=OFF
+          -DJPEG_LIBRARY=#{Formula["jpeg-turbo"].opt_lib}/libjpeg.so
+          -DOpenBLAS_LIB=#{Formula["openblas"].opt_lib}/libopenblas.so
+          -DOPENEXR_ILMIMF_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmImf.so
+          -DOPENEXR_ILMTHREAD_LIBRARY=#{Formula["openexr"].opt_lib}/libIlmThread.so
+          -DPNG_LIBRARY=#{Formula["libpng"].opt_lib}/libpng.so
+          -DPROTOBUF_LIBRARY=#{Formula["protobuf"].opt_lib}/libprotobuf.so
+          -DTIFF_LIBRARY=#{Formula["libtiff"].opt_lib}/libtiff.so
+          -DWITH_V4L=OFF
+          -DZLIB_LIBRARY=#{Formula["zlib"].opt_lib}/libz.so
+        ]
+      end
+
+      # Ref: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options
+      ENV.runtime_cpu_detection
+      if Hardware::CPU.intel? && build.bottle?
+        cpu_baseline = MacOS.version.requires_sse42? ? "SSE4_2" : "SSSE3"
+        args += %W[-DCPU_BASELINE=#{cpu_baseline} -DCPU_BASELINE_REQUIRE=#{cpu_baseline}]
+      end
+
+      system "cmake", "-S", ".", "-B", "build_shared", *args, *std_cmake_args
+      inreplace "build_shared/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
+      system "cmake", "--build", "build_shared"
+      system "cmake", "--install", "build_shared"
+
+      system "cmake", "-S", ".", "-B", "build_static", *args, *std_cmake_args, "-DBUILD_SHARED_LIBS=OFF"
+      inreplace "build_static/modules/core/version_string.inc", "#{Superenv.shims_path}/", ""
+      system "cmake", "--build", "build_static"
+      lib.install buildpath.glob("build_static/{lib,3rdparty/**}/*.a")
+
+      # clean build dirs before next iteration
+      rm_rf ["build_shared", "build_static"]
```

Comment:
Seems a bit excessive to build the full library 2 more times (doubling build time) and then discard majority of those builds just to add an extra Python binding to an already `long build` formula.

At least you definitely don't need another set of static libraries that aren't even related to the Python bindings.

Would be better to try to find/upstream a way to build separate Python bindings while reusing the existing shared C++ libs.

Reply:
I've tried to find a way to build the separate Python bindings; in fact, it _appears_ that you can do `cmake -S ./modules/python` to do just that! However, this "standalone Python bindings" mode requires that you've already built OpenCV, and building OpenCV requires passing it a reference to the Python executable. In practice, it appears that you can't use this "standalone Python bindings" mode to build the Python bindings for a _different version of Python_ than what you used for building OpenCV. 😞 

I've reduced this from 4 builds to 3, but I think that's the best I can do. If you (or anyone else) can find a better way of doing this, please let me know!